spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskDeleteService.java

```diff
@@ -193,7 +199,20 @@ private void performCleanupExecution(long id, String schemaTarget) {
 	@Override
 	@Transactional
 	public void cleanupExecutions(Set<TaskExecutionControllerDeleteAction> actionsAsSet, String taskName, boolean completed) {
-		List<AggregateTaskExecution> tasks = this.taskExplorer.findTaskExecutionsByName(taskName, completed);
+		cleanupExecutions(actionsAsSet, taskName, completed, null);
+	}
+
+	@Override
+	@Transactional
+	public void cleanupExecutions(Set<TaskExecutionControllerDeleteAction> actionsAsSet, String taskName, boolean completed, Integer days) {
+		List<AggregateTaskExecution> tasks;
+		if (days != null) {
```

Comment:
I think we could extract this code block to a DateUtils w/ a `numDaysAgoFromLocalMidnightToday(int n)`. It could also be used in the new method in DTES.

Reply:
Done.